include/flatbuffers/flatbuffers.h

```diff
@@ -75,6 +75,9 @@
 
 // The wire format uses a little endian encoding (since that's efficient for
 // the common platforms).
+#if defined(__s390x__)
+  #define FLATBUFFERS_LITTLEENDIAN 0
+#endif // __s390x__
 #if !defined(FLATBUFFERS_LITTLEENDIAN)
   #if defined(__GNUC__) || defined(__clang__)
     #ifdef __BIG_ENDIAN__
```

Comment:
By using ```#if defined(__BIG_ENDIAN__) || defined(__BIG_ENDIAN)``` code change, the following failure is seen on x86. 
flatc: /flatbuffers/include/flatbuffers/flatbuffers.h:171: void flatbuffers::EndianCheck(): Assertion `*reinterpret_cast<char *>(&endiantest) == FLATBUFFERS_LITTLEENDIAN' failed.

It seems on x86 __BIG_ENDIAN flag is recognised (even after setting __BYTE_ORDER to __LITTLE_ENDIAN) & sets define FLATBUFFERS_LITTLEENDIAN to '0'. As a result,  we see a failure in EndianCheck() function.

Hence, introduced the platform define to support s390x. Have added the flag above the existing ifdef so as to avoid breakage in the coding format.